serde/src/test/kotlin/com/amazon/ionhiveserde/objectinspectors/IonValueToStringObjectInspectorTest.kt

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazon.ionhiveserde.objectinspectors
+
+import com.amazon.ion.IonType
+import com.amazon.ion.IonValue
+import com.amazon.ion.impl._Private_Utils
+import com.amazon.ionhiveserde.ION
+import org.apache.hadoop.io.Text
+
+class IonValueToStringObjectInspectorTest : AbstractIonPrimitiveJavaObjectInspectorTest<IonValue, Text, String>() {
+
+    override val subject = com.amazon.ionhiveserde.objectinspectors.IonValueToStringObjectInspector()
+    override fun validTestCases() = listOf(
+            ValidTestCase(generateTestIonValues(IonType.STRING), "some string", Text("some string")),
+            ValidTestCase(generateTestIonValues(IonType.SYMBOL), "some string", Text("some string")),
+            ValidTestCase(generateTestIonValues(IonType.INT), "12", Text("12")),
+            ValidTestCase(generateTestIonValues(IonType.DECIMAL), "12.1", Text("12.1")),
+            ValidTestCase(generateTestIonValues(IonType.FLOAT), "12345e0", Text("12345e0")),
+            ValidTestCase(generateTestIonValues(IonType.BOOL), "true", Text("true")),
+            ValidTestCase(generateTestIonValues(IonType.TIMESTAMP), "2001-09-09T01:46:39.999Z", Text("2001-09-09T01:46:39.999Z")),
+            ValidTestCase(generateTestIonValues(IonType.STRUCT), "{field:1}", Text("{field:1}")),
+            ValidTestCase(generateTestIonValues(IonType.LIST), "[1]", Text("[1]")),
+            ValidTestCase(generateTestIonValues(IonType.SEXP), "(1)", Text("(1)")),
+            ValidTestCase(generateTestIonValues(IonType.CLOB), "{{\"test\"}}", Text("{{\"test\"}}")),
+            ValidTestCase(generateTestIonValues(IonType.BLOB), "{{AQIDBAU=}}", Text("{{AQIDBAU=}}")),
+    )
+
+    private fun generateTestIonValues(type: IonType): IonValue {
+        when (type) {
+            IonType.STRING -> {
+                return ION.newString("some string")
+            }
+            IonType.SYMBOL -> {
+                return ION.newSymbol("some string")
+            }
+            IonType.INT -> {
+                return ION.newInt(12)
+            }
+            IonType.DECIMAL -> {
+                return ION.newDecimal(12.1)
+            }
```

Comment:
This is not a good pattern- it hard-codes the values here and buries them. Instead of switching `generateTestIonValues` on IonType why not inline the values of these return statements?

Feedback applies to entire method.

Reply:
Got it, 

For some Ion type such as Ion struct and Ion List, we need extra lines to assign them values.

For example, we need below for Ion struct
```
IonType.STRUCT -> {
                val struct = ION.newEmptyStruct()
                struct.add("field", ION.newInt(1))
                return struct
            }
```

To make it clearer, I would create a method like `generateTextIonStruct(IonType, filed, value)` and pass the test data in its signature for these types that we need extra operations. For example, if we want to create a ionStruct `{a:2}` the method will just be `generateTextIonContainer(IonType.STRING, "a", 2)`. Let me know if I missed anything.